completed assessment

Lilmcg opened the pull request.

Lilmcg committed:
completed assessment

WalkerTexas-coder commented on code-challenges.test.js:
```diff
@@ -23,11 +23,39 @@ var secretCodeWord2 = "Gobbledygook"
+  })
+})
 
+// Ran test. Test failed. Error: leetCode is not defined
```
Perfect!

WalkerTexas-coder commented on code-challenges.test.js:
```diff
@@ -23,11 +23,39 @@ var secretCodeWord2 = "Gobbledygook"
+// console.log(leetCode(secretCodeWord2))
+// console.log(leetCode(secretCodeWord3))
+
+// Ran test again. Test passed!
```
Well done!

WalkerTexas-coder commented on code-challenges.test.js:
```diff
@@ -40,11 +68,32 @@ var arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+console.log(singleLetter(arrayOfWords1,letterA))
+console.log(singleLetter(arrayOfWords2,letterE))
+
+// Ran test again. Test passed!
```
Great job, good logic, great execution!

WalkerTexas-coder commented on code-challenges.test.js:
```diff
@@ -57,6 +106,61 @@ var hand2 = [5, 5, 3, 3, 4]
+
+// Ran test. Test passed!
+
+// Ignore this. I attempted using slice, splice, indexOf, includes based on soemthing I saw online...but did not get the pair incorporated
```
Thank you for the info! Feel free to delete any code you don't want to keep for your notes.

WalkerTexas-coder commented on code-challenges.test.js:
```diff
@@ -57,6 +106,61 @@ var hand2 = [5, 5, 3, 3, 4]
+  let vals = Object.values(countObject)
+  if((vals[0] === 2 && vals[1] === 3) || (vals[1] === 2 && vals[0] === 3)){
+    return true
+  }
```
Yahtzee!!!

WalkerTexas-coder commented on interview-questions.md:
```diff
@@ -7,53 +7,56 @@ Then, research the question to expand on your answer. Even if you feel you have
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: A relational database contains many tables (that have their data stored in rows and columns) that are linked to each other. Excel acts as a relational database. I used Excel for many years when I was a financial analyst and senior consultant. I was able to create columns and rows in Excel that housed all my data, from which I could extrapolate information to feed into my graphs, ch ...
```
This is the kind of answer that you want to hear in an interview!

WalkerTexas-coder commented on interview-questions.md:
```diff
@@ -7,53 +7,56 @@ Then, research the question to expand on your answer. Even if you feel you have
+  Your answer: A relational database contains many tables (that have their data stored in rows and columns) that are linked to each other. Excel acts as a relational database. I used Excel for many years when I was a financial analyst and senior consultant. I was able to create columns and rows in Excel that housed all my data, from which I could extrapolate information to feed into my graphs, ch ...
 
-  Researched answer:
+  Researched answer: A relational database contains tables made of rows and columns, and these tables have relationships to each other. Each row represents an instance of a class and the columns represent the instance variables. Excel and Google Sheets are examples of relational databases. Relational databases work well with structured data. Relational databases are also called SQL (structured qu ...
```
Also excellent!

WalkerTexas-coder commented on interview-questions.md:
```diff
@@ -7,53 +7,56 @@ Then, research the question to expand on your answer. Even if you feel you have
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys are a way to identify each row in a table. The key is unique for that instance of the class. There may be two Lillian (first name) McGillivray (last name) in a table, however Lillian McGillivray that lives in San Diego (city) will have a unique primary key that is different than Lillian McGillivray that lives in Austin (city). There would be no confusion regarding whic ...
```
Nicely done!

WalkerTexas-coder commented on ruby-challenges.rb:
```diff
@@ -19,16 +30,111 @@
 
+def sum_numbers(array)
+  array.sum
+end
```
Easy peasy!

WalkerTexas-coder commented on ruby-challenges.rb:
```diff
@@ -19,16 +30,111 @@
+p bike_one.pedal_faster(10)
+p bike_one.brake(25)
+
+# After I added and defined the values for wheels and current_speed in my initialize code line, it messed up line 94 and the error message was "wrong number of arguments". It was expecting 3 arguments after Bike.new  I already defined 'Trek' as the model within my get_model method (in question 3a), so now I am forced to pass 'Trek' (model) once again, along with the other 2 arguments (wheels and  ...
```
Your guess here is on point! This is not correct. We can check this idea by asking ourselves "do all bikes have a behavior that turns them into "trek" bikes? No, Trek is a brand that makes trek bikes. We have to tell this bike that it's being made by Trek. We do that by passing this information to it when the class is instantiated.

so your initialize method could (and I would argue should) look like this: 
```ruby
  def initialize(model)
      @model = model
      @wheels = 2
      @current_speed = 0
  end
  ```
  
  Because all bikes have 2 wheels, they all start at 0 speed and that is only changed by methods (think verbs). But bikes can come from any company-- but they all do come from some company so model has the be a req.

WalkerTexas-coder replied:
Still great work! If you hadn't gotten this far I wouldn't be of much help!

WalkerTexas-coder reviewed (approved):
Lillian you have some highly competent work here. The regex is impressive and your refactoring of your code is on point! I think it'll be helpful to practice making mock classes in your daily life-- think about how you would describe real world objects around you and what choices you would make in regards to them. I'd love to hear if this helps or not. Keep up the good work! This PR is approved!

Lilmcg commented:
Thanks Austin!

Lil

> On Oct 11, 2021, at 11:41 AM, Austin Walker ***@***.***> wrote:
> 
> ﻿
> @WalkerTexas-coder approved this pull request.
> 
> Lillian you have some highly competent work here. The regex is impressive and your refactoring of your code is on point! I think it'll be helpful to practice making mock classes in your daily life-- think about how you would describe real world objects around you and what choices you would make in regards to them. I'd love to hear if this helps or not. Keep up the good work! This PR is approved!
> 
> In code-challenges.test.js:
> 
> >  
> +// Ran test. Test failed. Error: leetCode is not defined
> Perfect!
> 
> In code-challenges.test.js:
> 
> >  
> +const leetCode = (string)=>{
> +  let stringSplit = string.split("")
> +  let newString = stringSplit.map(value => {
> +    return value = value.replace(/['a','A']/g,'4'),
> +    value = value.replace(/['e','E']/g,'3'),
> +    value = value.replace(/['i','I']/g,'1'),
> +    value = value.replace(/['o','O']/g,'0')
> +  }) return newString.join("")
> +}
> +
> +// console.log(leetCode(secretCodeWord1))
> +// console.log(leetCode(secretCodeWord2))
> +// console.log(leetCode(secretCodeWord3))
> +
> +// Ran test again. Test passed!
> Well done!
> 
> In code-challenges.test.js:
> 
> >  
> +const singleLetter = (array,string)=> {
> +    let newArray = array.filter(value => {
> +        return value.includes(string) || value.includes(string.toUpperCase())
> +      })
> +      return newArray
> +}
> +console.log(singleLetter(arrayOfWords1,letterA))
> +console.log(singleLetter(arrayOfWords2,letterE))
> +
> +// Ran test again. Test passed!
> Great job, good logic, great execution!
> 
> In code-challenges.test.js:
> 
> > +    countObject[x] = (countObject[x] || 0) + 1
> +  }
> +  let vals = Object.values(countObject)
> +  if((vals[0] === 2 && vals[1] === 3) || (vals[1] === 2 && vals[0] === 3)){
> +    return true
> +  }
> +  return false
> +}
> +
> +// console.log(fullHouse(hand1))
> +// console.log(fullHouse(hand2))
> +// console.log(fullHouse(hand3))
> +
> +// Ran test. Test passed!
> +
> +// Ignore this. I attempted using slice, splice, indexOf, includes based on soemthing I saw online...but did not get the pair incorporated
> Thank you for the info! Feel free to delete any code you don't want to keep for your notes.
> 
> In code-challenges.test.js:
> 
> > +// Create a function that takes in an array
> +// Create an object which holds the count of each value in the array
> +// Recognize two of the same value, or three of the same value
> +// Conditional if there are two of the same value and three of the same value
> +    // Return true
> +    // Otherwise return false
> +
> +const fullHouse = (array) => {
> +  let countObject = {}
> +  for(let x of array){
> +    countObject[x] = (countObject[x] || 0) + 1
> +  }
> +  let vals = Object.values(countObject)
> +  if((vals[0] === 2 && vals[1] === 3) || (vals[1] === 2 && vals[0] === 3)){
> +    return true
> +  }
> Yahtzee!!!
> 
> In interview-questions.md:
> 
> >  
>  
>  
>  4. What is a relational database? Are there other kinds of databases?
>  
> -  Your answer:
> +  Your answer: A relational database contains many tables (that have their data stored in rows and columns) that are linked to each other. Excel acts as a relational database. I used Excel for many years when I was a financial analyst and senior consultant. I was able to create columns and rows in Excel that housed all my data, from which I could extrapolate information to feed into my graphs, charts and scenario planning models. Whenever the data changed within those cells, my graphs, charts and scenario models would also adjust accordingly.
> This is the kind of answer that you want to hear in an interview!
> 
> In interview-questions.md:
> 
> >  
> -  Researched answer:
> +  Researched answer: A relational database contains tables made of rows and columns, and these tables have relationships to each other. Each row represents an instance of a class and the columns represent the instance variables. Excel and Google Sheets are examples of relational databases. Relational databases work well with structured data. Relational databases are also called SQL (structured query language) databases. Object-relational model is another kind of database. PostgreSQL is an example of this type of model. Non-relational databases include document-orientated databases, key-value stores, wide-column stores and graph stores. These are becoming more popular since not all data fits into a structured column/row set-up, and needs more flexibility.
> Also excellent!
> 
> In interview-questions.md:
> 
> >  
>  
>  
>  5. What are primary keys? Why are they important?
>  
> -  Your answer:
> +  Your answer: Primary keys are a way to identify each row in a table. The key is unique for that instance of the class. There may be two Lillian (first name) McGillivray (last name) in a table, however Lillian McGillivray that lives in San Diego (city) will have a unique primary key that is different than Lillian McGillivray that lives in Austin (city). There would be no confusion regarding which Lillian we are referring to thanks to the primary key.
> Nicely done!
> 
> In ruby-challenges.rb:
> 
> > @@ -19,16 +30,111 @@
>  nums_array2 = [25, 17, 47, 11]
>  # Expected output: 100
>  
> +def sum_numbers(array)
> +  array.sum
> +end
> Easy peasy!
> 
> In ruby-challenges.rb:
> 
> > +          @current_speed = 0
> +      else
> +          @current_speed -= reduce_speed
> +      end
> +  end
> +end
> +
> +bike_one = Bike.new("Trek", 2, 0)
> +bike_one.get_model
> +bike_one.bike_info
> +
> +p bike_one.pedal_faster(10)
> +p bike_one.pedal_faster(10)
> +p bike_one.brake(25)
> +
> +# After I added and defined the values for wheels and current_speed in my initialize code line, it messed up line 94 and the error message was "wrong number of arguments". It was expecting 3 arguments after Bike.new  I already defined 'Trek' as the model within my get_model method (in question 3a), so now I am forced to pass 'Trek' (model) once again, along with the other 2 arguments (wheels and current_speed) after Bike.new. Having the argument 'Trek' in both the get_model method and passing it through after Bike.new is probably not the best way to do this, but I'm leaving it this way in case it's actually ok to write the code like this.
> Your guess here is on point! This is not correct. We can check this idea by asking ourselves "do all bikes have a behavior that turns them into "trek" bikes? No, Trek is a brand that makes trek bikes. We have to tell this bike that it's being made by Trek. We do that by passing this information to it when the class is instantiated.
> 
> so your initialize method could (and I would argue should) look like this:
> 
>   def initialize(model)
>       @model = model
>       @wheels = 2
>       @current_speed = 0
>   end
> Because all bikes have 2 wheels, they all start at 0 speed and that is only changed by methods (think verbs). But bikes can come from any company-- but they all do come from some company so model has the be a req.
> 
> In ruby-challenges.rb:
> 
> > +          @current_speed = 0
> +      else
> +          @current_speed -= reduce_speed
> +      end
> +  end
> +end
> +
> +bike_one = Bike.new("Trek", 2, 0)
> +bike_one.get_model
> +bike_one.bike_info
> +
> +p bike_one.pedal_faster(10)
> +p bike_one.pedal_faster(10)
> +p bike_one.brake(25)
> +
> +# After I added and defined the values for wheels and current_speed in my initialize code line, it messed up line 94 and the error message was "wrong number of arguments". It was expecting 3 arguments after Bike.new  I already defined 'Trek' as the model within my get_model method (in question 3a), so now I am forced to pass 'Trek' (model) once again, along with the other 2 arguments (wheels and current_speed) after Bike.new. Having the argument 'Trek' in both the get_model method and passing it through after Bike.new is probably not the best way to do this, but I'm leaving it this way in case it's actually ok to write the code like this.
> Still great work! If you hadn't gotten this far I wouldn't be of much help!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android.
